Delete obsolete CheckEnforcer tool

konrad-jamrozik opened the pull request:
Deleting obsolete, no longer used sources of check enforcer, per this comment: https://github.com/Azure/azure-sdk-tools/pull/4937#pullrequestreview-1214475397. Now we use a different version of it, based on GitHub actions. 

Related work item:
- #4991

Once this PR is merged, I will remove the [internals / tool - check-enforcer](https://dev.azure.com/azure-sdk/internal/_build?definitionId=1671) pipeline as well.

@weshaggard @benbp I believe the README in the removed directory (`tools/check-enforcer/README`) probably has bunch of useful information and should be reclaimed/moved. I also want to use this as an opportunity to review and update our check enforcer docs. Hence my questions:

- Where should I move the README? To `doc` directory, and link to it from top-level README?
- Where is the currently active check enforcer source? Where is its documentation?

Thanks!

benbp commented:
FYI there is still one more repository using the old check enforcer, but I think it's ok to start deleting the code while we keep the service running. I have an update to migrate this repository [here](https://github.com/Azure/azure-dev/pull/981), but was blocked at the time on a minor issue. I will follow-up on this.

@konrad-jamrozik it is ok to delete the readme, the docs have been moved/updated and live [here](https://github.com/Azure/azure-sdk-actions/blob/main/docs/check-enforcer.md). The active check-enforcer source is in that same repo [here](https://github.com/Azure/azure-sdk-actions).

konrad-jamrozik commented:
> FYI there is still one more repository using the old check enforcer, but I think it's ok to start deleting the code while we keep the service running. I have an update to migrate this repository [here](https://github.com/Azure/azure-dev/pull/981), but was blocked at the time on a minor issue. I will follow-up on this.
> 
> @konrad-jamrozik it is ok to delete the readme, the docs have been moved/updated and live [here](https://github.com/Azure/azure-sdk-actions/blob/main/docs/check-enforcer.md). The active check-enforcer source is in that same repo [here](https://github.com/Azure/azure-sdk-actions).

@benbp for the source link, you linked just to the repo, but I guess you meant perhaps to link to the [handleEvent](https://github.com/Azure/azure-sdk-actions/blob/main/main.go#L68) method?

Also, please note that since you approved this PR I have updated the main README to point to the new Check Enforcer location, and I have removed the `CHECKENFORCER` file.

I will keep this PR open until we are fully migrated and we can decommission the tool. Just to avoid potential confusion on what is going on.

weshaggard commented on eng/CHECKENFORCER:
```diff
@@ -1,41 +0,0 @@
-format: v0.1-alpha
```
@benbp did you have any plans to use this configuration with the new checkenforcer? If not then we should probably remove this from the other repos too.

benbp replied:
No plans. I have an item on my todo to remove the other markdown files from the repos.

weshaggard replied:
Ok make sure your todo list translates to github issues so we don't lose them.

konrad-jamrozik replied:
@benbp I can go ahead and get rid of all the `CHECKENFORCER` files, given I already work on this.

Can I maybe also somehow assist with merging this?
https://github.com/Azure/azure-dev/pull/981

Just hoping here to wrap-up this removal work and not leave anything dangling.

konrad-jamrozik commented:
We are having discussion about this PR on internal Teams chat on `Engineering System / Azure SDK`, [here](https://teams.microsoft.com/l/message/19:59dbfadafb5e41c4890e2cd3d74cc7ba@thread.skype/1671146702316?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1671146702316&teamName=Azure%20SDK&channelName=Engineering%20System%20%F0%9F%9B%A0%EF%B8%8F&createdTime=1671146702316&allowXTenantAccess=false).

Looks it the new GitHub-action based Check Enforcer possibly cannot be triggered by other GitHub workflows, which makes it unsuitable for https://github.com/Azure/azure-dev, as it uses such workflows. We are currently discussing possible approaches.

konrad-jamrozik commented:
I am closing this PR for now, as there is more work to be done before we can decommission the tool. I captured our findings and work required here:
- #4991

Once the prerequisite work is done, we can re-open this PR.

konrad-jamrozik commented:
@benbp I have deleted the resource groups used by the legacy CheckEnforcer, per [this comment](https://github.com/Azure/azure-sdk-tools/issues/4991#issuecomment-1404604092). I believe now we can delete the tool itself, hence reopening this PR.

konrad-jamrozik commented:
/check-enforcer override
